src/odemis/acq/stream/_sync.py

```diff
@@ -1402,36 +1415,57 @@ def _integrateImages(self, n):
                 best_dtype = get_best_dtype_for_acc(orig_dtype, n)  # avoid saturation and overflow
                 data = numpy.sum(data2process, axis=0, dtype=best_dtype)
 
-                if model.MD_BASELINE in md:
-                    baseline = md[model.MD_BASELINE]
-                    # Subtract baseline (aka black level) to avoid it from being multiplied.
-                    # Remove baseline from n-1 images, keep one baseline as bg level.
-
-                    minv = float(data.min())
-
-                    # If the baseline is too high compared to the actual black, we
-                    # could end up subtracting too much, and values would underflow
-                    # => be extra careful and never subtract more than min value.
-                    baseline_sum = n * baseline  # sum of baselines for n images.
-                    extra_bl = (n - 1) * baseline  # sum of baselines for n-1 images, keep one baseline as bg level.
-                    # check if we underflow the data values
-                    if extra_bl > minv:
-                        extra_bl = minv
-                        logging.info("Baseline reported at %d * %d, but lower values found, so only subtracting %d",
-                                     baseline, n, extra_bl)
-
-                    # Same as "data -= extra_bl", but also works if extra_bl < 0
-                    numpy.subtract(data, extra_bl, out=data, casting="unsafe")
-                    md[model.MD_BASELINE] = baseline_sum - extra_bl
+            # replace images, which needed to be averaged/integrated, with averaged/integrated image
+            self._acq_data[stream_idx] = [model.DataArray(data, md)]
 
+    def _updateMetadata(self, n):
+        """
+        Update the metadata for the final integrated image which is saved in self._acq_data
+        Args:
+            n: (int) Number of images that need to be integrated (summed).
+        """
```

Comment:
Move it to `odemis.util.img`, with a more specific name, like `update_integration_metadata(md, n)`.